src/pages/ProductsPage.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from 'react';
+import Products from '../components/Products';
+import Sidebar from '../components/Sidebar';
+import ProductCategoriesData from '../mocks/en-us/product-categories.json';
+import { ProductPageContent } from './ProductPage.style';
+
+export default function ProductsPage() {
+  const [categoriesSelected, setCategoriesSelected] = useState([]);
+  const [categories, setCategories] = useState([]);
+
+  useEffect(() => {
+    let mounted = true;
+    const categoriesData = ProductCategoriesData.results.map((category) => {
+      return { ...category, isSelected: false };
+    });
+
+    if (mounted) setCategories(categoriesData);
+
+    return () => (mounted = false);
+  }, []);
```

Comment:
No creo que este useEffect sea necesario, podría estar fuera del componente y en tu declaración del state lo seteas como state inicial, aprovechando para renombrar la variable también, algo así:

```suggestion
import { ProductPageContent } from './ProductPage.style';

const initialCategories = ProductCategoriesData.results.map((category) => {
  return { ...category, isSelected: false };
});

export default function ProductsPage() {
  const [categoriesSelected, setCategoriesSelected] = useState([]);
  const [categories, setCategories] = useState(initialCategories);
```

Reply:
Tienes razón, es mejor inicializarlo de esta manera